### PR TITLE
Namespace changes

### LINKED ISSUE #285: Better namespaces
Following a comment from a user at spark summit today, I think we need to move to a better top-level namespace for our packages - e.g. `keystoneml.workflow` instead of just `workflow`

### PR DESCRIPTION
**Pull request by etrain**
This PR closes #285. I have added a top level "keystoneml" namespace and refactored accordingly. The trickier bits were making sure that the Makefiles and JNI bindings were up to date but after a `make` all tests pass locally.

### CODE REVIEW

File: src/test/scala/keystoneml/workflow/PipelineSuite.scala [hunk 1]
```diff
@@ -449,7 +449,7 @@ class PipelineSuite extends FunSuite with PipelineContext with Logging {
       }
     }, sc.parallelize(Seq(32, 94, 12)), sc.parallelize(Seq("10", "7", "14")))
 
-    // Execute pipelines 1 to 3
+    // Execute keystoneml.pipelines 1 to 3
```
Comment (shivaram) [inline src/test/scala/keystoneml/workflow/PipelineSuite.scala]: Not sure this one should change :)

File: src/test/scala/keystoneml/workflow/GraphSuite.scala [hunk 2]
```diff
@@ -736,7 +736,7 @@ class GraphSuite extends FunSuite with PipelineContext with Logging {
       graph.replaceNodes(nodesToRemove, graph2, replacementSourceSplice, replacementSinkSplice)
     }
 
-    // May only replace dependencies on removed nodes
+    // May only replace dependencies on removed keystoneml.nodes
```
Comment (shivaram) [inline src/test/scala/keystoneml/workflow/GraphSuite.scala]: All three uses of `nodes` don't need to change ?

File: src/main/scala/keystoneml/nodes/learning/BlockLinearMapper.scala [hunk 3]
```diff
@@ -26,7 +26,7 @@ class BlockLinearMapper(
     val featureScalersOpt: Option[Seq[Transformer[DenseVector[Double], DenseVector[Double]]]] = None)
   extends Transformer[DenseVector[Double], DenseVector[Double]] {
 
-  // Use identity nodes if we don't need to do scaling
+  // Use identity keystoneml.nodes if we don't need to do scaling
```
Comment (tomerk) [inline src/main/scala/keystoneml/nodes/learning/BlockLinearMapper.scala]: Looks like the find-and-replace caught a victim here

File: src/main/scala/keystoneml/nodes/util/Cacher.scala [hunk 4]
```diff
@@ -1,13 +1,13 @@
...
-import pipelines.Logging
-import workflow.Transformer
+import keystoneml.pipelines.Logging
+import keystoneml.workflow.Transformer
 
 import scala.reflect.ClassTag
 
 /**
- * Caches an RDD at a given point within a Pipeline. Follows Spark's lazy evaluation conventions.
+ * Caches an RDD at a given point within a Pipeline. Follows Spark's lazy keystoneml.evaluation conventions.
```
Comment (tomerk) [inline src/main/scala/keystoneml/nodes/util/Cacher.scala]: Another comment victim

File: src/main/scala/keystoneml/workflow/AnalysisUtils.scala [hunk 5]
```diff
@@ -1,9 +1,9 @@
-package workflow
+package keystoneml.workflow
 
 private[workflow] object AnalysisUtils {
 
   /**
-   * Given a graph and a source/sink/node, output the set of all sources/sinks/nodes
+   * Given a graph and a source/sink/node, output the set of all sources/sinks/keystoneml.nodes
```
Comment (tomerk) [inline src/main/scala/keystoneml/workflow/AnalysisUtils.scala]: Here too

File: src/main/scala/keystoneml/workflow/AnalysisUtils.scala [hunk 6]
```diff
@@ -26,7 +26,7 @@ private[workflow] object AnalysisUtils {
   }
 
   /**
-   * Given a graph and a source/sink/node, output the set of all sources/sinks/nodes
+   * Given a graph and a source/sink/node, output the set of all sources/sinks/keystoneml.nodes
```
Comment (tomerk) [inline src/main/scala/keystoneml/workflow/AnalysisUtils.scala]: and here

File: src/main/scala/keystoneml/workflow/AnalysisUtils.scala [hunk 7]
```diff
@@ -43,7 +43,7 @@ private[workflow] object AnalysisUtils {
   }
 
   /**
-   * Given a graph and a source/sink/node, output the set of all nodes
+   * Given a graph and a source/sink/node, output the set of all keystoneml.nodes
```
Comment (tomerk) [inline src/main/scala/keystoneml/workflow/AnalysisUtils.scala]: Basically everywhere in this file

File: src/main/scala/keystoneml/workflow/AutoCacheRule.scala [hunk 8]
```diff
@@ -50,7 +50,7 @@ class AutoCacheRule(cachingMode: CachingStrategy) extends Rule with Logging {
 
   /**
    * Get an estimate for how many times the output of each node will be accessed, assuming
-   * the given set of nodes have their outputs cached.
+   * the given set of keystoneml.nodes have their outputs cached.
```
Comment (tomerk) [inline src/main/scala/keystoneml/workflow/AutoCacheRule.scala]: This file has some too, but the occurrences here are mostly okay with `keystoneml.nodes` in the comments

File: src/main/scala/keystoneml/workflow/EquivalentNodeMergeRule.scala [hunk 9]
```diff
@@ -1,26 +1,26 @@
-package workflow
+package keystoneml.workflow
 
 /**
- * A rule to merge equivalent nodes in the DAG.
+ * A rule to merge equivalent keystoneml.nodes in the DAG.
```
Comment (tomerk) [inline src/main/scala/keystoneml/workflow/EquivalentNodeMergeRule.scala]: This file too, but also not the worst sinners.

**Comment by etrain** [comment]
Alright - made a comprehensive pass through the comments and removed instances of "keystoneml." I blame IntelliJ.

**Comment by tomerk** [comment]
LGTM now! Like @shivaram said we should probably update the website & example application repo after this.

**Comment by etrain** [comment]
Yup - I have created #294 to track this.

On Feb 27, 2017, at 12:56 PM, Tomer Kaftan <notifications@github.com> wrote:

LGTM now! Like @shivaram <https://github.com/shivaram> said we should
probably update the website & example application repo after this.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/amplab/keystone/pull/293#issuecomment-282850633>, or mute
the thread

**Comment by etrain** [comment]
Alright, going ahead and merging this.

**Pull request merged**